objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/neuralNet/YoloProcessor.java

```diff
@@ -291,6 +333,38 @@ public NeuralNetResults processImage(byte[] image, Map<String, ?> request) throw
                 }
             }
         }
+                
+        // Checking if pre cropping was specified in query parameters
+        boolean queryCrop = false;
+        int x, y, w, h;
+        x = y = w = h = -1;
+        if (request.get("preCrop") instanceof Map) {
+            Map preCrop = (Map) request.get("preCrop");
+            if (preCrop.get("x") instanceof Integer && (Integer) preCrop.get("x") >= 0) {
+                x = (Integer) preCrop.get("x");
+            }
+            if (preCrop.get("y") instanceof Integer && (Integer) preCrop.get("y") >= 0) {
+                y = (Integer) preCrop.get("y");
+            }
+            if (preCrop.get("w") instanceof Integer && (Integer) preCrop.get("w") >= 0) {
+                w = (Integer) preCrop.get("w");
+            }
+            if (preCrop.get("h") instanceof Integer && (Integer) preCrop.get("h") >= 0) {
+                h = (Integer) preCrop.get("h");
+            }
+            if (x >= 0 && y >= 0 && w >= 0 && h >= 0) {
```

Comment:
I don't think we'll work in [Flatland](https://en.wikipedia.org/wiki/Flatland), so I think width & height need to be > 1, don't they?